source/common/stats/stats_impl.h

```diff
@@ -333,27 +340,54 @@ class GaugeImpl : public Gauge, public MetricImpl {
   // Stats::Gauge
   virtual void add(uint64_t amount) override {
     data_.value_ += amount;
-    data_.flags_ |= RawStatData::Flags::Used;
+    data_.flags_ |= Flags::Used;
   }
   virtual void dec() override { sub(1); }
   virtual void inc() override { add(1); }
   virtual void set(uint64_t value) override {
     data_.value_ = value;
-    data_.flags_ |= RawStatData::Flags::Used;
+    data_.flags_ |= Flags::Used;
   }
   virtual void sub(uint64_t amount) override {
     ASSERT(data_.value_ >= amount);
     ASSERT(used());
     data_.value_ -= amount;
   }
   virtual uint64_t value() const override { return data_.value_; }
-  bool used() const override { return data_.flags_ & RawStatData::Flags::Used; }
+  bool used() const override { return data_.flags_ & Flags::Used; }
 
 private:
   RawStatData& data_;
   RawStatDataAllocator& alloc_;
 };
 
+/**
+ * Implementation of HistogramStatistics for circllhist.
+ */
+class HistogramStatisticsImpl : public HistogramStatistics {
+public:
+  HistogramStatisticsImpl() : computed_quantiles_(supported_quantiles_.size(), 0.0) {}
+  /**
+   * HistogramStatisticsImpl object is constructed using the passed in histogram.
+   * @param histogram_ptr pointer to the histogram for which stats will be calculated. This pointer
+   * will not be retained.
+   */
+  HistogramStatisticsImpl(const histogram_t* histogram_ptr);
+
+  HistogramStatisticsImpl(const HistogramStatisticsImpl&) = delete;
```

Comment:
Just use NonCopyable (https://github.com/envoyproxy/envoy/blob/master/source/common/common/non_copyable.h). Search code for examples of how to use.